core/option.go

```diff
@@ -13,3 +17,9 @@ func WithMetrics() Option {
 		p.metrics = true
 	}
 }
+
+func WithChainID(id p2p.Network) Option {
+	return func(p *params) {
+		p.chainID = id.String()
+	}
+}
```

Comment:
I'm curious about why it's optional. Could you share more about the scenarios where disabling chain name validation might be beneficial?

Reply:
Mostly so I didn't have to break constructor - I made a comment somewhere that i made it optional but we can also require it, I do not mind. This component is very temporary and hopefully to be removed in future.